File: programmers/java/src/level3/bfsdfs/네트워크.java
package level3.bfsdfs;
//https://school.programmers.co.kr/learn/courses/30/lessons/43162?language=java
public class 네트워크 {
    private int[][] computers;
    private int n;
    private boolean[] visited;
    public void dfs(int start) {
        visited[start] = true;
        for(int i=0; i<n; i++) {
            if(!visited[i] && computers[start][i] == 1) {
                dfs(i);
            }
        }

    }

    public int solution(int n, int[][] computers) {
        this.computers = computers;
        this.n = n;
        int answer = 0;
        visited = new boolean[n];
        for(int i=0; i<n; i++) {
            if(!visited[i]) {
                dfs(i);
                answer++;
            }
        }
        return answer;
    }
}

File: programmers/python/level3/dfs_bfs/네트워크.py
## https://school.programmers.co.kr/learn/courses/30/lessons/43162?language=python3
from collections import deque 
def solution(n, computers):
    
    candidates = [1] * n
    cnt = 0
    while 1 in candidates:
        now = candidates.index(1)
        cnt +=1 
        q = deque()
        q.append(now)
        
        while q:
            cur = q.popleft()
            candidates[cur] = 0
            for i in range(n):
                if computers[cur][i] == 1 and candidates[i] == 1:
                    q.append(i)
    return cnt
            
        
